homeassistant/components/mqtt/notify.py

```diff
@@ -161,20 +158,37 @@ async def async_get_service(
         )
         return None
     await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+
+    service_config: MqttNotificationConfig = MqttNotificationConfig(config)  # type: ignore
+    service_config[CONF_COMMAND_TEMPLATE] = MqttCommandTemplate(
+        config.get(CONF_COMMAND_TEMPLATE), hass=hass
+    )
     services[service_name] = MqttNotificationService(
         hass,
-        config[CONF_COMMAND_TOPIC],
-        MqttCommandTemplate(config.get(CONF_COMMAND_TEMPLATE), hass=hass),
-        config[CONF_ENCODING],
-        name,
-        config[CONF_QOS],
-        config[CONF_RETAIN],
-        config[CONF_TARGETS],
-        config[CONF_TITLE],
+        service_config,
     )
+    # Override the service name if name is not set
+    if CONF_NAME not in config:
+        config[CONF_NAME] = DOMAIN
     return services[service_name]
 
 
+class MqttNotificationConfig(TypedDict, total=False):
+    """Supply service parameters for MqttNotificationService."""
+
+    command_topic: str
+    command_template: MqttCommandTemplate
+    encoding: str
+    name: str | None
+    qos: int
+    retain: bool
+    targets: list
+    title: str
+    discovery_hash: tuple | None
+    device_id: str | None
+    config_entry: ConfigEntry | None
```

Comment:
Suggest not to include these since this is not included in the config

Reply:
done